module/optee/smt/test/mod_optee_smt_unit_test.c

```diff
@@ -0,0 +1,889 @@
+    channel_config =
+        (struct mod_optee_smt_channel_config*)smt_element_table[FAKE_SCMI_SERVICE_IDX_OSPM_0]
+            .data;
+
+   /* Save current type */
+    saved_type = channel_config->type;
+
+    /* Set invalid type */
+    channel_config->type = MOD_OPTEE_SMT_CHANNEL_TYPE_COUNT;
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = mailbox_channel_init(channel_id, slot_count, channel_config);
+    TEST_ASSERT_EQUAL(FWK_E_DATA, status);
+
+    /* Restore valid type */
+    channel_config->type = saved_type;
+
+    /* Save current address */
+    saved_address = channel_config->mailbox_address;
+
+    /* Set invalid mailbox address */
+    channel_config->mailbox_address = 0;
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = mailbox_channel_init(channel_id, slot_count, channel_config);
+    TEST_ASSERT_EQUAL(FWK_E_DATA, status);
+
+    /* Restore valid address */
+    channel_config->mailbox_address = saved_address;
+
+    /* Save current size */
+    saved_size = channel_config->mailbox_size;
+
+    /* Set invalid mailbox size */
+    channel_config->mailbox_size = 0;
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = mailbox_channel_init(channel_id, slot_count, channel_config);
+    TEST_ASSERT_EQUAL(FWK_E_DATA, status);
+
+    /* Restore valid size */
+    channel_config->mailbox_size = saved_size;
+
+    /* Fail to get virt address */
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    smt_phys_to_virt_ExpectAndReturn(channel_config->mailbox_address,
+                                     channel_config->mailbox_size,
+                                     channel_config->policies & MOD_SMT_POLICY_SECURE,
+                                     (uintptr_t)NULL);
+    status = mailbox_channel_init(channel_id, slot_count, channel_config);
+    TEST_ASSERT_EQUAL(FWK_E_DATA, status);
+
+    /* Successfully init channel */
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    smt_phys_to_virt_ExpectAndReturn(channel_config->mailbox_address,
+                                     channel_config->mailbox_size,
+                                     channel_config->policies & MOD_SMT_POLICY_SECURE,
+                                     (uintptr_t)mailbox_virt_buffer);
+    fwk_mm_alloc_ExpectAndReturn(1, channel_config->mailbox_size, channel_in_buffer);
+    fwk_mm_alloc_ExpectAndReturn(1, channel_config->mailbox_size, channel_out_buffer);
+    status = mailbox_channel_init(channel_id, slot_count, channel_config);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+}
+
+/*!
+ * \brief test module's bind fucntion for round 0.
+ *
+ *  \details Test bind for round 0 and others except for round 1 when it needs
+ *  additional information from bind_request of other modules.
+ */
+void test_mailbox_smt_bind_round0(void)
```

Comment:
The function name doesn't reflect this part. "and others except for round 1"
Can we say something like ..bind_rounds_except_round1?

Reply:
or maybe bind_trivial_rounds
anything indicate that it's not only round0 and they are of a specific nature